sycl/include/CL/sycl/queue.hpp

```diff
@@ -339,6 +374,32 @@ class __SYCL_EXPORT queue {
   /// \return an event representing fill operation.
   event memset(void *Ptr, int Value, size_t Count);
 
+  /// Fills the memory pointed by a USM pointer with the value specified.
+  /// No operations is done if \param Count is zero. An exception is thrown
+  /// if \param Ptr is nullptr. The behavior is undefined if \param Ptr
+  /// is invalid.
+  ///
+  /// \param Ptr is a USM pointer to the memory to fill.
+  /// \param Value is a value to be set. Value is cast as an unsigned char.
+  /// \param Count is a number of bytes to fill.
+  /// \param DepEvent is an event that specifies the kernel dependencies.
+  /// \return an event representing fill operation.
+  event memset(void *Ptr, int Value, size_t Count, event DepEvent);
+
+  /// Fills the memory pointed by a USM pointer with the value specified.
+  /// No operations is done if \param Count is zero. An exception is thrown
```

Comment:
This "no operation is done of Count is zero".  Is that check performed in the runtime? I didn't see that check.

Reply:
It is checked in [`MemoryManager::fill_usm`](https://github.com/intel/llvm/blob/82c915cc18717853921d9bfdcd9f95843230710a/sycl/source/detail/memory_manager.cpp#L664-L671), which is called by `queue_impl::memset`.

Reply:
thanks - I cruised right past that.